src/cascadia/TerminalApp/Command.cpp

```diff
@@ -11,15 +11,33 @@
 #include <LibraryResources.h>
 
 using namespace winrt::TerminalApp;
+using namespace winrt::Windows::Foundation;
 using namespace ::TerminalApp;
 
 static constexpr std::string_view NameKey{ "name" };
 static constexpr std::string_view IconPathKey{ "iconPath" };
 static constexpr std::string_view ActionKey{ "command" };
 static constexpr std::string_view ArgsKey{ "args" };
+static constexpr std::string_view IterateOnKey{ "iterateOn" };
+static constexpr std::string_view CommandsKey{ "commands" };
+
+static constexpr std::string_view IterateOnProfilesValue{ "profiles" };
+
+static constexpr std::wstring_view ProfileName{ L"${profile.name}" };
 
 namespace winrt::TerminalApp::implementation
 {
+    Command::Command()
+    {
+        _nestedCommandsView = winrt::single_threaded_vector<winrt::TerminalApp::Command>();

```

Comment:
should all commands have a vector allocated to store nested commands? That seems a little heavy

Reply:
Okay so technically, I bet this makes the most sense as just returning a `IMapView`. We'd have a `single_threaded_map<hstring, Command> _subcommands` in each `Command`, and `NestedCommands` would just return a view on those commands.

That leads to other complications though - `LayerJson` currently takes a `std::unordered_map`, so that would also have to be changed, which then would also mean that the map of commands in GlobalAppSettings should be an `IMap<String, Command>`. 

Considering the work @carlos-zamora is doing now with settings, this is probably the right long-run call, it's just gonna take me another day 😝

Reply:
ah shoot I'm gonna have to pull the commands out of the map to build a temporary vector for the actual palette at some point though. Shoot. Too bad there's not an `IIterable<V> IMap<K, V>::GetValues()`

Reply:
Okay, it's done now.